src/app/services/product.service.ts

```diff
@@ -0,0 +1,47 @@
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs';
+import {HttpClient} from "@angular/common/http"
+import {ProductItemDetailed} from "../interfaces/ProductItemDetailed";
+import {Order} from "../interfaces/Order";
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class ProductService {
+
+  orders: Order [] = []
+
+  constructor(private httpClient: HttpClient) {
+  }
+
+  getOrders() {
+    return this.orders
+  }
+
+  getProduct(id: number): Observable<ProductItemDetailed> {
+    return this.httpClient.get<ProductItemDetailed>("http://localhost:3000/products/" + id);
```

Comment:
**suggestion:** you cand keep this base url inside the `environment.ts` file or you could use a [proxy](https://angular.io/guide/build#proxying-to-a-backend-server)